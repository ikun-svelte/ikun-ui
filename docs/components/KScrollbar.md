---
title: KScrollbar
lang: en-US
---

# KScrollbar

A scroll bar dom container with a unified style

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/scrollbar
```

```bash [yarn]
yarn add @ikun-ui/scrollbar
```

```bash [npm]
npm install @ikun-ui/scrollbar
```

:::

## Basic usage

<demo src="scrollbar/basic.svelte"  github='Scrollbar'></demo>

## Customizable style

scrollbar supports partial style settings for `tumb`, `bar`, and `track`

<demo src="scrollbar/custom.svelte" github='Scrollbar'></demo>

## Scrollbar Props

| Name            | Type     | Default                 | Description                                                |
| --------------- | -------- | ----------------------- |------------------------------------------------------------|
| css             | `string` | `string`                | Css content which will be appended to the style attribute. |
| trackBackground | `string` | `transparent`           | Scrollbar track background color.                          |
| trackRadius     | `string` | `20px`                  | Scrollbar track radius.                                    |
| width           | `string` | `6px`                   | Scrollbar width.                                           |
| height          | `string` | `6px`                   | Scrollbar height.                                          |
| thumbBackground | `string` | `var(--ikun-light-800)` | Thumb background color.                                    |
| thumbRadius     | `string` | `20px`                  | Thumb background radius.                                   |
| cls             | `string` | `-`                     | Additional class for component                             |
| attrs           | `any`    | `{}`                    | Additional attributes                                      |

## Scrollbar Slots

| Name    | Description               |
| ------- | ------------------------- |
| default | content of the container. |
