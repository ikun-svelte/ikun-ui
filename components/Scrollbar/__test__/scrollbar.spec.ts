import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import KScrollbar from '../src';

let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.useRealTimers();
});

describe('Test: KScrollbar', () => {
	test('props: cls', async () => {
		const instance = new KScrollbar({
			target: host,
			props: {
				cls: 'k-scrollbar--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-scrollbar--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('basic render', async () => {
		const instance = new KScrollbar({
			target: host,
			props: {
				cls: 'k-scrollbar--test',
				width: '10px',
				height: '10px',
				thumbBackground: 'red',
				thumbRadius: '50px',
				trackRadius: '50px',
				trackBackground: 'green'
			}
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
