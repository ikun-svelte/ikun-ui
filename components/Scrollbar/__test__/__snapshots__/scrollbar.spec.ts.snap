// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KScrollbar > basic render 1`] = `"<div class=\\"k-scrollbar k-scrollbar--test\\" style=\\"--k-scrollbar--thumb__radius: 50px; --k-scrollbar--thumb: red; --k-scrollbar__h: 10px; --k-scrollbar__w: 10px; --k-scrollbar--track__radius: 50px; --k-scrollbar--track: green;\\"></div>"`;

exports[`Test: KScrollbar > props: cls 1`] = `"<div class=\\"k-scrollbar k-scrollbar--test\\" style=\\"--k-scrollbar--thumb__radius: 20px; --k-scrollbar--thumb: var(--ikun-light-800); --k-scrollbar__h: 6px; --k-scrollbar__w: 6px; --k-scrollbar--track__radius: 20px; --k-scrollbar--track: transparent;\\"></div>"`;
